dojo/importers/importer/importer.py

```diff
@@ -179,18 +179,32 @@ def close_old_findings(self, test, scan_date_time, user, push_to_jira=None):
         # Would it make more sense to exclude duplicates? But the deduplication process can be unfinished because it's
         # run in a celery async task...
         if test.engagement.deduplication_on_engagement:
-            old_findings = Finding.objects.exclude(test=test) \
-                                            .exclude(hash_code__in=new_hash_codes) \
-                                            .filter(test__engagement=test.engagement,
-                                                test__test_type=test.test_type,
-                                                active=True)
+            if service:
+                old_findings = Finding.objects.exclude(test=test) \
+                                                .exclude(hash_code__in=new_hash_codes) \
+                                                .filter(test__engagement=test.engagement,
+                                                    test__test_type=test.test_type,
+                                                    active=True, service=service)
+            else:
+                old_findings = Finding.objects.exclude(test=test) \
+                                                .exclude(hash_code__in=new_hash_codes) \
+                                                .filter(test__engagement=test.engagement,
+                                                    test__test_type=test.test_type,
+                                                    active=True)
         else:
             # TODO BUG? this will violate the deduplication_on_engagement setting for other engagements
-            old_findings = Finding.objects.exclude(test=test) \
-                                            .exclude(hash_code__in=new_hash_codes) \
-                                            .filter(test__engagement__product=test.engagement.product,
-                                                test__test_type=test.test_type,
-                                                active=True)
+            if service:
+                old_findings = Finding.objects.exclude(test=test) \
+                                                .exclude(hash_code__in=new_hash_codes) \
+                                                .filter(test__engagement__product=test.engagement.product,
+                                                    test__test_type=test.test_type,
+                                                    active=True, service=service)
+            else:
+                old_findings = Finding.objects.exclude(test=test) \
+                                                .exclude(hash_code__in=new_hash_codes) \
+                                                .filter(test__engagement__product=test.engagement.product,
+                                                    test__test_type=test.test_type,
+                                                    active=True)
```

Comment:
```suggestion
            old_findings = Finding.objects.exclude(test=test) \
                                                .exclude(hash_code__in=new_hash_codes) \
                                                .filter(test__engagement__product=test.engagement.product,
                                                    test__test_type=test.test_type,
                                                    active=True)

            if service:
                old_findings = old_findings.filter(service=service)
```

Reply:
Thank you @valentijnscholten, your suggestion is not only better but made me find a bug as well. The new version works with all combinations of having or not having a service when importing scans.

Reply:
What's the 'service=True' for?

Reply:
This was a mistake of course, should have filtered for `is_null` what it's doing now. That was the reason why the unit test for API's initially failed.

Reply:
@valentijnscholten Added some test cases